data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/producer/JsonLineProducerTask.java

```diff
@@ -0,0 +1,127 @@
+  }
+
+  /**
+   * Process ScalarDB scan result data and returns CSV data
+   *
+   * @param dataChunk list of results
+   * @return result converted to string
+   */
+  @Override
+  public String process(List<Result> dataChunk) {
+    StringBuilder jsonLines = new StringBuilder();
+
+    for (Result result : dataChunk) {
+      ObjectNode objectNode = generateJsonForResult(result);
+      jsonLines.append(objectNode.toString());
+      jsonLines.append(System.lineSeparator());
+    }
+    return jsonLines.toString();
+  }
+
+  /**
+   * Generate a Json Object based on a ScalarDB Result
+   *
+   * @param result ScalarDB Result object instance
+   * @return JsonObject containing the ScalarDB result data
+   */
+  private ObjectNode generateJsonForResult(Result result) {
+    LinkedHashSet<String> tableColumns = tableMetadata.getColumnNames();
+
+    ObjectNode objectNode = objectMapper.createObjectNode();
+
+    // Loop through all the columns and to the json object
+    for (String columnName : tableColumns) {
+      // Skip the field if it can be ignored based on check
+      boolean columnNotProjected = !projectedColumnsSet.contains(columnName);
+      boolean isMetadataColumn =
+          ConsensusCommitUtils.isTransactionMetaColumn(columnName, tableMetadata);
+      if (columnNotProjected || (!includeMetadata && isMetadataColumn)) {
+        continue;
+      }
+
+      DataType dataType = dataTypeByColumnName.get(columnName);
+      addToObjectNode(objectNode, result, columnName, dataType);
+    }
+    return objectNode;
+  }
+
+  /**
+   * Add result column name and value to json object node
+   *
+   * @param result ScalarDB result
+   * @param columnName column name
+   * @param dataType datatype of the column
+   */
+  private void addToObjectNode(
+      ObjectNode objectNode, Result result, String columnName, DataType dataType) {
+
+    if (result.isNull(columnName)) {
+      return;
+    }
+
+    switch (dataType) {
+      case BOOLEAN:
+        objectNode.put(columnName, result.getBoolean(columnName));
+        break;
+      case INT:
+        objectNode.put(columnName, result.getInt(columnName));
+        break;
+      case BIGINT:
+        objectNode.put(columnName, result.getBigInt(columnName));
+        break;
+      case FLOAT:
+        objectNode.put(columnName, result.getFloat(columnName));
+        break;
+      case DOUBLE:
+        objectNode.put(columnName, result.getDouble(columnName));
+        break;
+      case TEXT:
+        objectNode.put(columnName, result.getText(columnName));
+        break;
+      case BLOB:
+        // convert to base64 string
+        byte[] encoded = Base64.getEncoder().encode(result.getBlobAsBytes(columnName));
+        objectNode.put(columnName, new String(encoded, Charset.defaultCharset()));
+        break;
+      default:
+        logger.error(
+            CoreError.DATA_LOADER_CONVERT_TO_STRING_FAILED.buildMessage(dataType.toString()));
+        break;
```

Comment:
Diito. We can throw an `AssertionError` since this code should never be reached.